client/my-sites/hosting/phpmyadmin-card/restore-db-password.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getHttpData, httpData, requestHttpData } from 'state/data-layer/http-data';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import Dialog from 'components/dialog';
+import { connect } from 'react-redux';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { successNotice } from 'state/notices/actions';
+
+const requestId = siteId => `restore-db-password-${ siteId }`;
+
+export const restoreDatabasePassword = siteId =>
+	requestHttpData(
+		requestId( siteId ),
+		http(
+			{
+				method: 'POST',
+				path: `/sites/${ siteId }/hosting/restore-database-password`,
+				apiNamespace: 'wpcom/v2',
+				body: {},
+			},
+			{}
+		),
+		{
+			fromApi: () => () => [ [ requestId( siteId ), null ] ],
+			freshness: 0,
+		}
+	);
+
+const RestorePasswordDialog = ( {
+	isVisible,
+	onRestore,
+	onCancel,
+	isRestoring,
+	hasRestored,
+	siteId,
+	translate,
+	showSuccessNotice,
+	reset,
+} ) => {
+	if ( hasRestored ) {
+		reset();
+		showSuccessNotice( translate( 'Your database password has been restored.' ), {
+			duration: 3000,
+		} );
+		onRestore();
+	}
+
+	const buttons = [
+		{
+			action: 'restore',
+			label: translate( 'Restore' ),
+			onClick: () => restoreDatabasePassword( siteId ),
+			isPrimary: true,
+			disabled: isRestoring,
+			additionalClassNames: isRestoring ? 'is-busy' : '',
+		},
+		{
+			action: 'cancel',
+			label: translate( 'Cancel' ),
+			onClick: onCancel,
+			disabled: isRestoring,
+		},
+	];
+	return (
+		<Dialog
+			isVisible={ isVisible }
+			buttons={ buttons }
+			onClose={ onCancel }
+			shouldCloseOnEsc={ ! isRestoring }
+		>
+			<h1>{ translate( 'Restore database password' ) }</h1>
+			<p>
+				{ translate( 'Are you sure you want to restore the default password of your database?' ) }
```

Comment:
I'm not sure how long this request should take - it's pretty snappy locally - but it might worth be throwing in a loading state (even just on the button for this dialog).

Reply:
Already handled. We disable both buttons and add the `is-busy` class to the restore button when the request is pending.

Reply:
Already handled. We disable both buttons and add the is-busy class to the restore button when the request is pending.

Reply:
Finally decided to drop that behavior and automatically hide the dialog. We'll show a notification when the request finishes. I see we apply the same pattern in other parts (trashing a post, muting a conversation, etc) without showing any loading indicator.